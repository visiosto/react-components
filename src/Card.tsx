// Copyright (c) 2021 Visiosto oy
// Licensed under the MIT License

import React from 'react';
import styled from 'styled-components';

import boxShadowHoverStyle from './style/boxShadowHoverStyle';
import boxShadowStyle from './style/boxShadowStyle';

const Article = styled.article<{ centered?: boolean; smallerPadding?: boolean }>`
  border-radius: 0.5rem;
  padding: ${(props) => (props.smallerPadding ? 0 : '2rem')};
  transition: box-shadow 100ms ease-in;
  text-align: ${(props) => (props.centered ? 'center' : 'left')};

  ${boxShadowStyle}

  &:hover {
    ${boxShadowHoverStyle}
  }
`;

export type CardProps = { centered?: boolean; children: React.ReactNode; smallerPadding?: boolean };

function Card({ centered = false, children, smallerPadding = false }: CardProps) {
  return (
    <Article centered={centered} smallerPadding={smallerPadding}>
      {children}
    </Article>
  );
}

export default Card;
