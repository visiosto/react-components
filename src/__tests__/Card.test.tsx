// Copyright (c) 2021 Visiosto oy
// Licensed under the Apache License, Version 2.0

import React from 'react';
import { screen } from '@testing-library/react';

import Card from '../Card';

import renderWithProviders from '../util/test/renderWithProviders';

describe('Card component', () => {
  it('renders correctly', () => {
    const { container, getByText } = renderWithProviders(<Card>Test text</Card>);

    expect(container).toMatchSnapshot();

    expect(getByText('Test text')).toBeInTheDocument();

    expect(screen.getByRole('article')).toHaveTextContent('Test text');
  });

  it('renders centred card correctly', () => {
    const { container, getByText } = renderWithProviders(<Card centered>Test text</Card>);

    expect(container).toMatchSnapshot();

    expect(getByText('Test text')).toBeInTheDocument();

    expect(screen.getByRole('article')).toHaveTextContent('Test text');
  });

  it('renders card with smaller padding correctly', () => {
    const { container, getByText } = renderWithProviders(<Card smallerPadding>Test text</Card>);

    expect(container).toMatchSnapshot();

    expect(getByText('Test text')).toBeInTheDocument();

    expect(screen.getByRole('article')).toHaveTextContent('Test text');
  });

  it('renders centred card with smaller padding correctly', () => {
    const { container, getByText } = renderWithProviders(
      <Card centered smallerPadding>
        Test text
      </Card>,
    );

    expect(container).toMatchSnapshot();

    expect(getByText('Test text')).toBeInTheDocument();

    expect(screen.getByRole('article')).toHaveTextContent('Test text');
  });
});
