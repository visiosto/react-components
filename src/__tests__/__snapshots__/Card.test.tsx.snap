// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders card with smaller padding correctly 1`] = `
<div>
  <article
    class="sc-bdvvtL hNslzD"
  >
    Test text
  </article>
</div>
`;

exports[`Card component renders centred card correctly 1`] = `
<div>
  <article
    class="sc-bdvvtL hUOIrv"
  >
    Test text
  </article>
</div>
`;

exports[`Card component renders centred card with smaller padding correctly 1`] = `
<div>
  <article
    class="sc-bdvvtL gqJgFf"
  >
    Test text
  </article>
</div>
`;

exports[`Card component renders correctly 1`] = `
<div>
  <article
    class="sc-bdvvtL escuhn"
  >
    Test text
  </article>
</div>
`;
